### ISSUE TITLE
carbonapi 0.12.3 segfault?

### ISSUE THREAD

**Issue by reyjrar** [go-graphite/carbonapi#407]
I have 4 nodes running carbonapi.  I'm integration testing and looking to replace the graphite-web as the "Default" backend for Grafana.  To do that, I enabled the `graphiteWeb()` functionality on one node to test the `forceAdd` to route `integralByInterval()` calls to the local graphite-web instance.  

While I was sleeping, it segfaulted on an unused function (I am not using tags in graphite or carbonapi at all, and have *no* queries using tags):

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xc3f04e]

goroutine 25767 [running]:
github.com/go-graphite/carbonapi/zipper/errors.(*Errors).Add(...)
        /builddir/golang/src/github.com/go-graphite/carbon
api/zipper/errors/errors.go:75
github.com/go-graphite/carbonapi/zipper/protocols/v3.(*ClientProtoV3Group).doTagQuery(0xc000166480, 0xf1c020, 0xc0004467c0, 0x1, 0x0, 0x0, 0
xffffffffffffffff, 0x0, 0x0, 0x0, ...)
        /builddir/golang/src/github.com/go-graphite/carbon
api/zipper/protocols/v3/protobuf_group.go:247 +0x2ce
github.com/go-graphite/carbonapi/zipper/protocols/v3.(*ClientProtoV3Group).TagNames(0xc000166480, 0xf1c020, 0xc0004467c0, 0x0, 0x0, 0xffffff
ffffffffff, 0x26, 0xc00344efc0, 0xd16d60, 0x1)
        /builddir/golang/src/github.com/go-graphite/carbon
api/zipper/protocols/v3/protobuf_group.go:259 +0x6f
github.com/go-graphite/carbonapi/zipper/broadcast.(*BroadcastGroup).doTagRequest(0xc000466120, 0xf1c020, 0xc0004467c0, 0xc000502900, 0xf270a
0, 0xc000166480, 0xd70920, 0xc0005b8660, 0xc000502840)
        /builddir/golang/src/github.com/go-graphite/carbon
api/zipper/broadcast/broadcast_group.go:461 +0x553
github.com/go-graphite/carbonapi/zipper/broadcast.(*BroadcastGroup).doTagRequest-fm(0xf1c020, 0xc0004467c0, 0xc000502720, 0xf270a0, 0xc00016
6480, 0xd70920, 0xc0005b8660, 0xc000502840)
        /builddir/golang/src/github.com/go-graphite/carbon
api/zipper/broadcast/broadcast_group.go:493 +0x83
created by github.com/go-graphite/carbonapi/zipper/types.DoRequest
        /builddir/golang/src/github.com/go-graphite/carbon
api/zipper/types/response.go:43 +0x224
```

The other nodes did not and the only difference is the `graphiteWeb` enabling in the configs:

```yaml
#  to maintain backward compatibility, this function is disabled by default
enabled: true
# roundrobin will be used in case multiple URLs are specified
fallbackUrls:
    - http://localhost:80
# if enabled, function will also check if supported parameters are the same
strict: false
# specify max amount of connections per each instance of graphit-web
maxConcurrentConnections: 10
# specify request timeouts
timeout: "90s"
keepAliveInterval: "30s"
# Uncomment to make those functions force-proxied to graphite-web
forceAdd:
    - "integralByInterval"
#     - "sumSeries"
#     - "derivative"

# This will remove those functions from list of proxied ones
# This is example and also suggestion for functions to remove.
forceSkip:
    - "aliasByTags"
    - "seriesByTag"
    - "groupByTags"
    - "events"
```

**Comment by Civil** [comment]
Grafana still might do queries for tags. And tag support for carbonapi_v3_pb protocol is untested, as the only software that supports that protocol is go-carbon, which doesn't support tags.

I was trying to rework all error handling across the zipper in this branch: https://github.com/go-graphite/carbonapi/tree/zipper_improve_errorhandling

Could you please test building it?

For the current master the fix is easy and it's commited to master (in case you don't have time to test the new branch)

**Comment by ralphM78** [comment]
Hello,

I have noticed the same error with carbonapi 0.12.5 and carbonapi_v2_pb.

**Comment by Civil** [comment]
Could you provide panic logs and config? Without that I cannot really debug the issue.

**Comment by ralphM78** [comment]
Setup is Grafana 6.2.5 -> carbonapi-0.12.5 -> graphite-clickhouse

Config is this example only with other server addresses
https://github.com/go-graphite/carbonapi/blob/master/cmd/carbonapi/carbonapi.example.clickhouse.yaml

goroutine 433 [running]:
github.com/go-graphite/carbonapi/zipper/errors.(*Errors).Add(...)
/home/go/src/carbonapi/zipper/errors/errors.go:75
github.com/go-graphite/carbonapi/zipper/protocols/v2.(*ClientProtoV2Group).doTagQuery(0xc0002d2600, 0x100dbc0, 0xc00084bd00, 0x1, 0x0, 0x0, 0xffffffffffffffff, 0x0, 0x0, 0x0, ...)
/home/go/src/carbonapi/zipper/protocols/v2/protobuf_group.go:351 +0x2d0
github.com/go-graphite/carbonapi/zipper/protocols/v2.(*ClientProtoV2Group).TagNames(0xc0002d2600, 0x100dbc0, 0xc00084bd00, 0x0, 0x0, 0xffffffffffffffff, 0x22, 0x0, 0x0, 0x0)
/home/go/src/carbonapi/zipper/protocols/v2/protobuf_group.go:363 +0x6f
github.com/go-graphite/carbonapi/zipper/broadcast.(*BroadcastGroup).doTagRequest(0xc00030c000, 0x100dbc0, 0xc00084bd00, 0xc00a2fa420, 0x101eaa0, 0xc0002d2600, 0xe0a320, 0xc0001ab540, 0xc00a2fa3c0)
/home/go/src/carbonapi/zipper/broadcast/broadcast_group.go:461 +0x5af
created by github.com/go-graphite/carbonapi/zipper/types.DoRequest
/home/go/src/carbonapi/zipper/types/response.go:43 +0x224
carbonapi.service: main process exited, code=exited, status=2/INVALIDARGUMENT
Unit carbonapi.service entered failed state.
systemd[1]: carbonapi.service failed.

I have tested your fix #407 with carbonapi_v2_pb and it semms to work.
